### DISCUSSION TITLE
@click triggers twice, nothing I do prevents this

### DISCUSSION THREAD

**Discussion by JF-626164** [alpinejs/alpine#3803] [1. Help] [answered]
Hello!

As the title suggests, I'm encountering an issue where the code I have contained in the @click is triggering twice. I've tried using .debounce, .prevent, and .once modifiers to prevent this but none worked. I learned after using .once that it really means the button its attached is only allowed to be clicked once.

Any ideas how to make it trigger the desired code only once?

Thanks!!
<hr>
<h1>Code:</h1>
I'm working on a TALL stack, Alpinejs version: 3.13.1

<h2>Blade code:</h2>

```
// Triggering code:
<button x-ignore class="btn btn-primary mt-2 mr-3 mb-4 text-white float-right" @click="updateData()">Update Data</button>
<button class="btn btn-primary mt-2 mr-3 mb-4 text-white float-right" @click.once="$dispatch('testing')">test</button>

// Alpine x-data "updateData()" function
updateData() {
        console.log('inside alpine function');
}

// In the script tag
window.addEventListener('testing', event => {
        console.log('it worked');
});
```

<h2>Output:</h2>
Inside the browser console:

```
// testing
it worked
it worked

// updateData()
inside alpine function
inside alpine function
```

**Comment by trych** [comment]
Is it possible that you're initializing Alpine twice?

**Comment by ekwoka** [comment]
You've definitely called Alpine.start twice.

**Comment by JF-626164** [comment, marked as answer]
@trych @ekwoka, yall were right. I had alpine started twice. 

Some advice for the next person who has this issue; I'm using Livewire v3 with my repo, v3 ships with AlpineJs in it, so there is no need for you to import AlpineJs seperately. Likewise v3 already runs Alpine.start() so there is no need to do it manually. Just make sure you are following the Livewire Installation [documentation](https://livewire.laravel.com/docs/installation#manually-bundling-livewire-and-alpine) closely.
